### ISSUE TITLE
Consider using `.Dockerfile` as a extension

### ISSUE THREAD

**Issue by nstogner** [ardanlabs/service#15]
The `.Dockerfile` extension provides better editor-friendliness.

i.e. `metrics.Dockerfile` instead of `dockerfile.metrics`

**Comment by ardan-bkennedy** [comment]
If we use the current convention, all the dockerfiles stay together in the tree. Isn't that better?

**Comment by wyattjoh** [comment]
Maybe a good "editor-friendly" change would be to move each Dockerfile along side each of the binaries that it's associated with?

```
cmd/crud/Dockerfile
cmd/sidecar/metrics/Dockerfile
cmd/sidecar/tracer/Dockerfile
```

**Comment by ardan-bkennedy** [comment]
I had it like that orginially and ran into problems. :( I don't remember what those problems were.

**Comment by nstogner** [comment]
@ardan-bkennedy I agree that it looks bad to have them scattered in `ls` output. Since they do pertain to a given binary I think @wyattjoh might be on to something, however I am hesitant to put no-go files in `cmd/`

**Comment by nstogner** [comment]
@ardan-bkennedy It might have to related to the build context?

**Comment by ardan-bkennedy** [comment]
I was having trouble with location related stuff when building. I think we should close this and leave it as is?

**Comment by wyattjoh** [comment]
The issue before was build context's yeah.

Something like a build script (or gasp, Makefile) could make calls like:

```sh
docker build -f cmd/crud/Dockerfile -t ardanlabs/service-crud .
docker build -f cmd/sidecar/metrics/Dockerfile -t ardanlabs/service-metrics .
docker build -f cmd/sidecar/tracer/Dockerfile -t ardanlabs/service-tracer .
```

Or what have you. The idea is that the Dockerfile would be co-located with the primary binary.

**Comment by ardan-bkennedy** [comment]
Right now I am happy with all the build artifacts being at the root of the project.

**Comment by ardan-bkennedy** [comment]
I would like to close this issue unless someone else objects.
